src/app/lib/utilities/preprocessor/rules/blockPositioning.js

```diff
@@ -0,0 +1,49 @@
+import deepGet from '../../../../helpers/json/deepGet';
+import deepClone from '../../../../helpers/json/deepClone';
+
+const insertBlockPositioning = (
+  { model },
+  positionArr = [],
+) => {
+  if (!model || typeof model !== 'object') {
+    return null;
+  }
+
+  const amendedJson = model;
+  let incrementPositionAtLevel = 0;
+  Object.keys(model).forEach(key => {
+    if (key !== 'blocks') {
+      return null;
+    }
+
+    return model[key].map((childObj, index) => {
+      const newPosition = [...positionArr, (incrementPositionAtLevel += 1)];
+      const newChildObj = Object.assign(childObj, { position: newPosition });
```

Comment:
```suggestion
      childObj.position = newPosition;
```

If I've understood correctly, we already do a deep clone before manipulating the data so ideally we don't want to be making new objects for every block.

Reply:
iirc eslint will throw a `no-param-reassign` error. Will confirm in the morning.

Reply:
Yep above linter error is thrown.